### PR TITLE
Updating outdated links

### PR DESCRIPTION
**Pull request by murchu27**
Updating the links to the docs and source code for `ast.rs`. Seems like it was moved to a new crate at some point.

EDIT: also updating some links in the README ([comment below](https://github.com/rust-lang/rustfmt/pull/4869#issuecomment-860965546))

### CODE REVIEW

**Comment by murchu27** [comment]
Pushed [31b789d](https://github.com/rust-lang/rustfmt/pull/4869/commits/31b789d0ba7195cafe4f2aaa12e5e4dee4dbf1f4) to update some links in the README, too. Specifically the links to the `fmt-rfcs` reposiory, which appears to have changed owner

File: Contributing.md [hunk 1]
```diff
@@ -138,8 +138,8 @@ format.
 
 There are different nodes for every kind of item and expression in Rust. For
 more details see the source code in the compiler -
-[ast.rs](https://dxr.mozilla.org/rust/source/src/libsyntax/ast.rs) - and/or the
-[docs](https://doc.rust-lang.org/nightly/nightly-rustc/syntax/ast/index.html).
+[ast.rs](https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_ast/ast.rs.html) - and/or the
```
Comment (calebcartwright) [inline Contributing.md:141]: Probably better to point this one directly to source

```suggestion
[ast.rs](https://github.com/rust-lang/rust/blob/master/compiler/rustc_ast/src/ast.rs) - and/or the
```

**Comment by murchu27** [comment]
@calebcartwright you mention in a [comment on another PR ](https://github.com/rust-lang/rustfmt/pull/4865#issuecomment-859197717) that you want to avoid merge commits.

> it'd be best to submit the pulls from feature branches that have are up to date/rebased against the trunk branch here

Do the changes in this PR allow for this? Or do I need to change anything?

**Comment by calebcartwright** [comment]
> @calebcartwright you mention in a [comment on another PR ](https://github.com/rust-lang/rustfmt/pull/4865#issuecomment-859197717) that you want to avoid merge commits.
> 
> > it'd be best to submit the pulls from feature branches that have are up to date/rebased against the trunk branch here
> 
> Do the changes in this PR allow for this? Or do I need to change anything?

This is fine, thanks for asking though! It's primarily about how your source branch for the PR is updated, when necessary, and doing so by rebasing.

I can always squash all the commits as part of landing a PR, but there are times when it's helpful to maintain the individual commits which isn't an option if the PR source branch has merge commits. (related note, going to squash these as we don't need 3 commits for a 4 line diff)

**Pull request merged**